### DISCUSSION TITLE
Working with dates in SQLite

### DISCUSSION THREAD

**Discussion by PabloRuiz-TT** [Azure/azure-mobile-apps#784] [Q&A]
I require a query where the current date is validated with my dateStart and dateEnd fields in my local table, my query is as follows: 

JObject definition = new()
{
    { "workTeamId", string.Empty },
    { "start", default(DateTime) }
};

DateTime currentDate = DateTime.Now; ( 2023-10-11 )

var query =
    "select workTeamId, start " +
    "from workTeamDetail " +
    $"where start <= '{currentDate}' and finish >= '{currentDate}'";

var result = await _store.ExecuteQueryAsync(definition, query);

when I only compare with start, it returns the record but when I compare with start and finish it returns empty.

My record that was synchronized has the following in SQL Server:
Field start => 2023-10-07 00:00:00.0000000 (datetime)
Field Finish => 2023-10-13 00:00:00.0000000 (datetime)

My model is represented as follows
public DateTime Start { get; set; }
public DateTime Finish { get; set; }

I have tried the query in the same way in SQLite online and it does not give me any problems, any ideas?

**Comment by adrianhall** [comment]
When you do a ExecuteQueryAsync(), you are actually passing the SQL to SQLite, then transforming the result.  I'd need to trace this to be able to say definitively what is going wrong.

The best idea is to submit a PR with an xUnit test (in the appropriate project) that fails.  We can then step through the code and see what is actually going wrong.  

That being said, there is no reason why you can't just use the standard LINQ queries here - there is nothing unusual here.  Why are you using ExecuteQueryAsync?
